packages/dom/src/dom/is-text-field.js

```diff
@@ -23,7 +28,11 @@ export default function isTextField( element ) {
 		'number',
 	];
 	return (
-		( nodeName === 'INPUT' && ! nonTextInputs.includes( element.type ) ) ||
+		( element.nodeName === 'INPUT' &&
+			/** @type {HTMLInputElement} */ ( element ).type &&
```

Comment:
Does it not detect the `element.nodeName === 'INPUT'` check? What if you check `typeof element === HTMLInputElement`?

Reply:
I get this error when I try to use `typeof`:

`This condition will always return 'false' since the types 'string' and '{ new (): HTMLInputElement; prototype: HTMLInputElement; }' have no overlap.`

`eslint` also complains that `HTMLInputElement` is undefined 😞 I'll add a type guard?

Reply:
Well I could just convert it to TypeScript and the inline comments would go away 😝 

I'll do the type guard though, we should have something like that anyhow I think, it'll be useful in other cases as well I'm sure.